documentation/proposals/Proposal - Project Governance, Bindings Expansion and Ownership.md

```diff
@@ -0,0 +1,142 @@
+**could**, and **optional** are to be interpreted as described in RFC 2199 [1]. The additional key word **optionally**
+is an alternate form of **optional**, for use where grammatically appropriate. These key words are highlighted in the
+proposal for clarity.
+
+# Proposal
+
+## Problem Statement
+
+Silk.NET currently has a very high bar for bindings to be created under the Silk.NET umbrella. This is problematic because, although not for lack of trying [2], BuildTools was designed too heavily in tandem with Silk.NET (and to a small extent its predecessor OpenTK 4.0 for the period in which Silk.NET maintainers were responsible for its development) and did not scale well to non-Khronos and/or out-of-tree bindings. Despite the efforts in establishing the Silk.NET Community Program [3], obtaining bindings to other tangentially-related libraries that currently do not meet the Silk.NET team's criteria for inclusion under the Silk.NET umbrella remains problematic for users that are looking for the same level of reliability and/or trust vested in the Silk.NET brand as an extension of it being under the .NET Foundation umbrella. To this end, this proposal aims to:
+
+1. Define our current criteria as best as we can.
+2. Create an ownership and maintenance model that allows for expansion of our criteria for inclusion.
+3. Codify the new criteria against the 3.0 development plan [4].
+
+## Today's Criteria
+
+We never formally wrote down the criteria for inclusion in Silk.NET prior to 3.0 (the "Current Criteria") so this criteria is based on the bindings portfolio as of 2.20.0 and the current consensus on the maintenance team with regards to our perception of what such criteria would look like and what we have rejected in the past.
+
+In order for a binding to be included in the then-currently maintained version of Silk.NET prior to 3.0, all of the following criteria **must** be met:
+
+1. The C API being bound to **must** have open-source headers that do not preclude the automated transformation to C# bindings without undue licensing provisions inflicted on users of such transformations. 
+2. The C API being bound to **must** have at least one open-source implementation that is usable for the vast majority of use cases of that binding.
+    - The Silk.NET team has in the past allowed implementations with copyleft restrictions (namely with regards to static linking a la LGPL) that don't apply on desktop but would apply on mobile.
+3. The C API **must** be a recognised industry standard multimedia or compute API, aid in the creation of applications making use of such APIs, and/or used to implement a Silk.NET High Level Utility.
+    - The "recognised industry standard multimedia or compute API" covers all Khronos APIs, OpenAL, and the 2.X portfolio of DirectX bindings.
+    - The "aid in the creation of applications making use of such APIs" covers Assimp and Shaderc.
+    - The "used to implement a Silk.NET High Level Utility" covers GLFW and SDL.
```

Comment:
I understand why you write this, but seeing that Silk SDL bindings are the best ones available, is it there for the HLU or primarily for Standard Multimedia API?

Reply:
Basically, we will maintain our SDL bindings to the best of our ability so long as we're using them in our windowing/input HLU APIs. This does not preclude the bindings going beyond what is needed for the HLUs e.g. binding to the rendering APIs for instance, but it sets a tone for how we approach their maintenance.

This is no different to how we approach the SDL bindings in 2.X today.

Reply:
So if the HLU, which some might not use (preferring their own), switches to direct Mac/windows/X bindings, Silk SDL could be dropped according to the new proposal.

Or rather, it would have to become an Additional Binding.

According to https://www.nuget.org/profiles/Silk.NET, the most popular bindings are OGL/Vulkan, then LWGL, and then SDL. The HLU is actually less popular than SDL itself. Do I read that correctly?

Reply:
I think you're correct that we should likely make it so core bindings must remain core for the entirety of the major version's lifecycle.

Reply:
I've encompassed this in the proposal.